c10/core/TensorImpl.h

```diff
@@ -1056,6 +1056,24 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     }
   }
 
+  /**
+   * Whether or not the tensor is a zerotensor
+  */
+  inline bool is_zerotensor() const {
+    return key_set_.has(DispatchKey::ZeroTensor);
+  }
+
+  /**
+   Set whether or not the tensor is a zero tensor
+  */
+  void _set_zero(bool value) {
```

Comment:
In which context do you use this function?

Reply:
to not propagate the ZT bit in `_like` factory functions